test-d/readonly-deep.ts

```diff
@@ -0,0 +1,44 @@
+import {expectType, expectError} from 'tsd';
+import {ReadonlyDeep} from '../source/readonly-deep';
+
+const data = {
+	object: {
+		foo: 'bar'
+	},
+	fn: (_: string) => true,
+	string: 'foo',
+	number: 1,
+	boolean: false,
+	symbol: Symbol('test'),
+	null: null,
+	undefined: undefined, // eslint-disable-line object-shorthand
+	map: new Map<string, string>(),
+	set: new Set<string>(),
+	array: ['foo'],
+	tuple: ['foo'] as ['foo'],
+	readonlyMap: new Map<string, string>() as ReadonlyMap<string, string>,
+	readonlySet: new Set<string>() as ReadonlySet<string>,
+	readonlyArray: ['foo'] as ReadonlyArray<string>,
+	readonlyTuple: ['foo'] as const
```

Comment:
Can/should we handle values in a Promise?

See: https://gist.github.com/masterkidan/7322752f569b1bba53e0426266768623

Reply:
I don't think that we should handle promises here.

Reply:
Alright